src/module_2/explicacion_data_analysis.md

```diff
@@ -0,0 +1,425 @@
+
+En el segundo vemos que en el mes de enero ha subido mucho el numero de orders ( en los dos ultimos años ), que el primer año fue lineal el crecimiento de el numero de orders y vemos una gran caida en marzo de 2022 pero al ser el ultimo mes seguramente sea porque no ha finalizado todavia y quedan futuros pedidos por contabilizar
+
+En el ultimo grafico vemos pico de abandono a las 8 ,10 y 14. Asi de primeras se me ocurre que puede ser por que el cliente esta haciendo su compra justo antes de otra actividad importante (irse a trabajar , comer) por lo que se le echa el tiempo encima y deja a medias su order.
+
+
+```python
+ # Columnas a verificar
+ columns_to_check = ['count_people', 'count_adults', 'count_children', 'count_babies', 'count_pets']
+ df_cleaned = drop_rows_where_all_null(df_users, columns_to_check)
+ 
+ print(count_nulls(df_users,'count_people'))
+ print(df_users['count_people'].size)
+
+ print(count_nulls(df_cleaned,'count_people'))
+ print(df_cleaned['count_people'].size)
+ 
+```
+
+    4658
+    4983
+    0
+    325
+
+
+aqui podemos ver que de todo nuestro dataset de 4983 usuarios solo 325 completaron algun campo acerca de los miembros de su familia, por ende solo usaremos los que si estan completos para su posterior estudio
+
+
+
+```python
+df_families_members_stats()
+```
+
+
+    
+![png](explicacion_data_analysis_files/explicacion_data_analysis_22_0.png)
+    
+
+
+    Total de familias (excluyendo nulos): 322
+    Familias con adultos: 322
+    Familias con niños: 129
+    Familias con bebés: 22
+    Familias con mascotas: 129
+
+
+Notese que habia 3 casos de familias sin un adulto que han sido consideradas como errores ya que una mascota no ha hecho ningun pedido.
+Excluyendo estos 3 casos vemos muchas familias tienen niños y mascotas pero no tantas tienen bebes
+Con esta informacion podriamos aumentar el numero de productos para estos grupos y no añadir productos para bebes a nuestro inventario
+
+
+```python
+  # Código para generar el gráfico de recuento de orders por vendor
+vendor_order_counts = df_orders_combined_inventory.groupby('vendor')['number_of_orders'].sum().sort_values(ascending=False)
+plt.figure(figsize=(15, 8))  # Aumentamos el tamaño de la figura
+vendor_bars = plt.bar(vendor_order_counts.index, vendor_order_counts.values, color='skyblue')
+plt.xlabel('Vendor')
+plt.ylabel('Número de Orders')
+plt.title('Número de Orders por Vendor')
+plt.xticks(rotation=90)  # Rotamos las etiquetas 90 grados para evitar superposición
+for bar in vendor_bars:
+    height = bar.get_height()
+    plt.text(bar.get_x() + bar.get_width()/2, height, int(height), ha='center', va='bottom')
+plt.tight_layout()  # Ajustamos el layout para que no cortemos las etiquetas
+plt.show()
+```
+
+
+    
+![png](explicacion_data_analysis_files/explicacion_data_analysis_24_0.png)
```

Comment:
ou cannot see anything here. Everything is cluttered with text. You could just consider top X to have a zoom  of the relevant part and have a clean plot